db/indexes.go

```diff
@@ -384,6 +384,9 @@ func removeObsoleteIndexes(bucket base.N1QLBucket, previewOnly bool, useXattrs b
 	for _, sgIndex := range sgIndexes {
 		// Current version, opposite xattr setting
 		removalCandidates = append(removalCandidates, sgIndex.fullIndexName(!useXattrs))
+		if useViews {
+			removalCandidates = append(removalCandidates, sgIndex.fullIndexName(useXattrs))
```

Comment:
Should remove both xattr and non-xattr indexes when using views.

Reply:
Is this not covered by the line just outside of the if statement?

Reply:
Ah yeah. You're right.

I wonder if it would be clearer like this, to match below previous version logic:

```golang
if useViews {
	// Current version, both xattr and non-xattr
	removalCandidates = append(removalCandidates, sgIndex.fullIndexName(true))
	removalCandidates = append(removalCandidates, sgIndex.fullIndexName(false))
} else {
	// Current version, opposite xattr setting
	removalCandidates = append(removalCandidates, sgIndex.fullIndexName(!useXattrs))
}
```

Reply:
Ah yeah, you're right.

I wonder if it would be clearer like this, to match below previous version logic:

```golang
if useViews {
	// Current version, both xattr and non-xattr
	removalCandidates = append(removalCandidates, sgIndex.fullIndexName(true))
	removalCandidates = append(removalCandidates, sgIndex.fullIndexName(false))
} else {
	// Current version, opposite xattr setting
	removalCandidates = append(removalCandidates, sgIndex.fullIndexName(!useXattrs))
}
```

Reply:
Not really sure this is worth doing. I've added a comment to hopefully make it more clear. If you think its really necessary I can switch it to your suggestion though.

Reply:
The comment is enough. Thanks